src/brevitas/quant/solver/common.py

```
@@ -142,28 +151,41 @@ def bit_width_impl(bit_width_impl_type):
+    def scaling_stats_impl(scaling_stats_op=None, restrict_scaling_type=None):
         if scaling_stats_op is None:
-            return None
-        if scaling_stats_op == StatsOp.MAX:
-            return AbsMax
+            scaling_stats_impl = None
+        elif scaling_stats_op == StatsOp.MAX:
+            scaling_stats_impl = AbsMax
         elif scaling_stats_op == StatsOp.MAX_AVE:
-            return AbsMaxAve
+            scaling_stats_impl = AbsMaxAve
         elif scaling_stats_op == StatsOp.AVE:
-            return AbsAve
+            scaling_stats_impl = AbsAve
         elif scaling_stats_op == StatsOp.MEAN_SIGMA_STD:
-            return MeanSigmaStd
+            scaling_stats_impl = MeanSigmaStd
         elif scaling_stats_op == StatsOp.MEAN_LEARN_SIGMA_STD:
-            return MeanLearnedSigmaStd
+            scaling_stats_impl = MeanLearnedSigmaStd
         elif scaling_stats_op == StatsOp.PERCENTILE:
-            return AbsPercentile
+            scaling_stats_impl = AbsPercentile
         elif scaling_stats_op == StatsOp.MIN_MAX:
-            return AbsMinMax
+            scaling_stats_impl = AbsMinMax
         elif scaling_stats_op == StatsOp.PERCENTILE_INTERVAL:
-            return PercentileInterval
+            scaling_stats_impl = PercentileInterval
+        elif scaling_stats_op == StatsOp.SIGNED_MAX:
+            scaling_stats_impl = SignedAbsMax
         else:
             raise RuntimeError(f"{scaling_stats_op} not recognized.")
 
+        # For power of two scales, the stat needs to be unsigned
+        if restrict_scaling_type == RestrictValueType.POWER_OF_TWO and scaling_stats_op in [
+                StatsOp.SIGNED_MAX]:
+            warnings.warn(
+                f"Statistic {scaling_stats_impl.__name__} is signed, which is incompatible with the restriction to "
+                f"power of twos, so its absolute value will be taken. Consider switching to an unsigned statistic."
+            )
```

Didn't we say that this should just raise an error?
The user is responsible to pick the correct stats

---

I changed it to raise a `ValueError` in this scenario.